### DISCUSSION TITLE
Option To Rewrite Supported Links to Privacy Respective Frontends

### DISCUSSION THREAD

**Discussion by seniorm0ment** [Docile-Alligator/Infinity-For-Reddit#927] [Ideas]
Requesting the option to be able to have all YouTube links redirect to Invidious, Maps to OpenStreetMaps, Twitter to Nitter, Reddit to Teddit, Instagram to Bibliogram, etc.

These are all privacy respective frontends, for their respective services.

An example use would be, if while reading a comment someone posts a Twitter link, Infinity rewrites it as a Nitter link.

If someone posts a YouTube link, Infinity rewrites it as an Invidious link.

Fedilab does this, as do a few other apps.
This would of course be optional, and the user would be able to select their instances of choice.

**Comment by LivingWithHippos** [comment]
At the moment I'm using UntrackMe to do this if anyone want to try this stuff (it's on f droid)

**Comment by seniorm0ment** [comment]
This is different than rewriting the links, but also it is nice to not need to go through another app.
I can imagine it is trivial to add a toggle to rewrite links like such.

**Comment by LivingWithHippos** [comment]
The advantage is that this app works with every other app in your phone, and if you check  "use this every time" you aren't even going to notice it`s in use.

Also what you call "rewriting a link¨  would work exactly like this.

**Comment by seniorm0ment** [comment]
Last I used that app, anytime upon clicking on a recognized link, it opened up a redirect menu to click to open the link instead of simply redirecting and opening the link.
Even when set to always use.

I just tried it again, and that seems to be the case still, however it is also just crashing after clicking to open the link. So idk.

Also when I say rewriting a link, it would work like such however it would be contained within the app, negating the need for another app to do the same thing.

Also rewriting != redirecting in my suggestion. Although the same result goal.
